src/containers/collaborations/components/RemoveDataSetFromCollaborationModal.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * @flow
+ */
+
+import React from 'react';
+
+import { Map, get } from 'immutable';
+import { CollaborationsApiActions } from 'lattice-sagas';
+import { ActionModal, Typography } from 'lattice-ui-kit';
+import { ReduxUtils, useRequestState } from 'lattice-utils';
+import { useDispatch } from 'react-redux';
+import { RequestStates } from 'redux-reqseq';
+import type { UUID } from 'lattice';
+import type { RequestState } from 'redux-reqseq';
+
+import { ModalBody, ResetOnUnmount } from '../../../components';
+import { COLLABORATIONS } from '../../../core/redux/constants';
+import {
+  ID,
+  NAME,
+  ORGANIZATION_ID,
+  TITLE
+} from '../../../utils/constants';
+
+const { isSuccess } = ReduxUtils;
+
+const { REMOVE_DATA_SET_FROM_COLLABORATION, removeDataSetFromCollaboration } = CollaborationsApiActions;
+
+const RESET_ACTIONS = [REMOVE_DATA_SET_FROM_COLLABORATION];
+
+const RemoveDataSetFromCollaborationModal = ({
+  collaborationId,
+  dataSet,
+  isVisible,
+  onClose
+} :{
+  collaborationId :UUID;
+  dataSet :Map;
+  isVisible :boolean;
+  onClose :() => void;
+}) => {
+  const dispatch = useDispatch();
+  const dataSetId = get(dataSet, ID);
+  const dataSetName = get(dataSet, NAME);
+  const dataSetTitle = get(dataSet, TITLE);
+  const organizationId = get(dataSet, ORGANIZATION_ID);
+  const removeDataSetRS :?RequestState = useRequestState([COLLABORATIONS, REMOVE_DATA_SET_FROM_COLLABORATION]);
+
+  const success = isSuccess(removeDataSetRS);
+
+  const rsComponents = {
+    [RequestStates.STANDBY]: (
+      <ModalBody>
+        <span>
+          {`Are you sure you want to remove the data set ${dataSetTitle || dataSetName} from this collaboration?`}
+        </span>
+      </ModalBody>
+    ),
+    [RequestStates.SUCCESS]: (
+      <ResetOnUnmount actions={RESET_ACTIONS}>
+        <Typography>{`${dataSetTitle || dataSetName} was successfully removed from the collaboration.`}</Typography>
+      </ResetOnUnmount>
+    ),
+    [RequestStates.FAILURE]: (
+      <ResetOnUnmount actions={RESET_ACTIONS}>
+        <Typography>Failed to remove the data set. Please try again.</Typography>
+      </ResetOnUnmount>
+    ),
```

Comment:
I wonder if closing during pending fails to reset. Could be a problem for this pattern we've used throughout the app actually...

Reply:
hmm this is a good question.